src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1008,17 +1008,25 @@ DQLiftGatherSymCW::DQLiftGatherSymCW() {
         const auto& matched_out_gather = node_to_output.at(gather);
 
         // Create new gathers on W and S, connect respectively
-        auto new_cvt_w = std::make_shared<ov::op::v0::Convert>(matched_out_w, ov::element::f16);
+        auto new_cvt_w = std::make_shared<ov::op::v0::Convert>(matched_out_w, matched_out_s.get_element_type());
         auto gather_c = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, 0);
         auto new_g_w = std::make_shared<ov::op::v8::Gather>(new_cvt_w, matched_out_ids, gather_c);
         auto new_g_s = std::make_shared<ov::op::v8::Gather>(matched_out_s, matched_out_ids, gather_c);
         auto new_mul = std::make_shared<ov::op::v1::Multiply>(new_g_w, new_g_s);
-        auto new_out = std::make_shared<ov::op::v0::Convert>(new_mul, ov::element::f32);
 
         // Reconnect old gather readers to the new Multiply
-        for (auto&& r : matched_out_gather.get_target_inputs()) {
-            r.replace_source_output(new_out);
+        auto qcvtm_iter = node_to_output.find(qcvtm);
+        if (qcvtm_iter != node_to_output.end()) {
+            auto new_out = std::make_shared<ov::op::v0::Convert>(new_mul, qcvtm_iter->second.get_element_type());
+            for (auto&& r : matched_out_gather.get_target_inputs()) {
+                r.replace_source_output(new_out);
+            }
+        } else {
+            for (auto&& r : matched_out_gather.get_target_inputs()) {
+                r.replace_source_output(new_mul);
+            }
         }
+
```

Comment:
There's no description in the ticket, what's the reason for this change?

Reply:
Eugene is right